app/controllers/hiring_staff/sign_in/dfe/sessions_controller.rb

```diff
@@ -11,47 +11,82 @@ def new
   end
 
   def create
-    Rails.logger.warn("Hiring staff signed in: #{oid}")
-
-    permissions = TeacherVacancyAuthorisation::Permissions.new
-    permissions.authorise(identifier, selected_school_urn)
-
-    log_succesful_authentication
-
-    if permissions.authorised?
-      update_session(permissions.school_urn, permissions)
-      redirect_to school_path
+    Rails.logger.warn("Hiring staff signed in: #{user_id}")
+    if DfeSignInAuthorisationFeature.enabled?
+      perform_dfe_sign_in_authorisation
     else
-      not_authorised
+      perform_non_dfe_sign_in_authorisation
+    end
+  rescue Authorisation::ExternalServerError => error
+    Rollbar.log(:error, error)
+    respond_to do |format|
+      format.html { render 'errors/external_server_error', status: :server_error }
     end
   end
 
   private
 
   def not_authorised
-    log_failed_authorisation
+    audit_failed_authorisation
+    Rails.logger.warn("Hiring staff not authorised: #{user_id} for school: #{school_urn}")
+
     @identifier = identifier
     render 'user-not-authorised'
   end
 
-  def update_session(school_urn, permissions)
-    session.update(session_id: oid, urn: school_urn, multiple_schools: permissions.many?)
-    log_succesful_authorisation
+  def update_session
+    session.update(session_id: user_id, urn: school_urn)
+    audit_successful_authorisation
+  end
+
+  def update_non_dsi_session(school_urn, permissions)
```

Comment:
We've got some duplication here, will be nice to remove this as part of "Only show the "Change organisation" link when the logged in user is invited into multiple organisations"

Reply:
Agreed, will look to remove this as part of that